daemon/src/close_cfds.rs

```diff
@@ -0,0 +1,48 @@
+use crate::db;
+use async_trait::async_trait;
+use std::time::Duration;
+use tokio_tasks::Tasks;
+use xtra_productivity::xtra_productivity;
+use xtras::SendInterval;
+
+/// Interval at which we move closed CFDs to the correct database
+/// table.
+const CLOSE_CFDS_INTERVAL: Duration = Duration::from_secs(5 * 60);
+
+pub struct Actor {
+    db: db::Connection,
+    tasks: Tasks,
+}
+
+impl Actor {
+    pub fn new(db: db::Connection) -> Self {
+        Self {
+            db,
+            tasks: Tasks::default(),
+        }
+    }
+}
+
+#[async_trait]
+impl xtra::Actor for Actor {
+    type Stop = ();
+
+    async fn started(&mut self, ctx: &mut xtra::Context<Self>) {
+        let this = ctx.address().expect("we are alive");
+        self.tasks
+            .add(this.send_interval(CLOSE_CFDS_INTERVAL, || CloseCfds));
+    }
+
+    async fn stopped(self) -> Self::Stop {}
+}
+
+#[xtra_productivity]
+impl Actor {
+    async fn handle(&mut self, _: CloseCfds) {
+        if let Err(e) = self.db.move_to_closed_cfds().await {
```

Comment:
Hmm, my understanding is that this actor shouldn't block the rest of the application by itself.

It may have some effect on the ability for other parts of the codebase to access the database (I'm not exactly sure how sqlite behaves with concurrent access, but that is something that we had to contend with before this PR anyway), but I don't think spawning separate tasks would make a difference. That is because there are no other handlers in `close_cfds::Actor` that could be blocked. Theoretically it could block itself if the previous call to `move_to_closed_cfds` took longer than 5 minutes, but that would be surprising.

Reply:
fair enough, I think I was not thinking it through in depth when I posted my comment.